framework/src/imgui_manager.cpp

```diff
@@ -62,56 +62,40 @@ namespace gvk
 		// engine should expose them. ImageCount lets Dear ImGui know how many framebuffers and resources in general it should
 		// allocate. MinImageCount is not actually used even though there is a check at init time that its value is greater than 1.
 		// Source: https://frguthmann.github.io/posts/vulkan_imgui/
-	    init_info.MinImageCount = std::max(static_cast<uint32_t>(2u), std::max(static_cast<uint32_t>(wnd->number_of_frames_in_flight()), static_cast<uint32_t>(wnd->number_of_swapchain_images())));
+	    init_info.MinImageCount = 2u;
```

Comment:
Not sure if it is really related to this change.... probably not because `MinImageCount` could never be lower than `2u`, but I'll post the problem here anyways: 

On an AMD GPU, a ImGui-internal assert fails, namely the following:
``` 
IM_ASSERT(info->ImageCount >= info->MinImageCount);
```
in `imgui_impl_vulkan.cpp`, line #796.

This happens when both, the "#Concurrent Frames" and the "#Presentable Images" are set to `1`.

I don't know if it is relevant here, but just to make sure: The number of swapchain images _requested_ does _not_ have to match the number of swapchain images _actually provided_ by the GPU. They can be lower (maybe also higher?) than the number that was requested.

Reply:
We have another problem with the AMD GPU, namely a validation error that comes from the inside of ImGui:
```
ERR:  Debug utils callback with Id[1214025378|VUID-vkFreeMemory-memory-00677] and Message[Validation Error: [ VUID-vkFreeMemory-memory-00677 ] Object 0: handle = 0x20a72876330, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x485c8ea2 | Cannot call vkFreeMemory on VkDeviceMemory 0x34b11c0000001142[] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to memory (via images or buffers) must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.162.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeMemory-memory-00677)]
22: gvk::get_current_callstack (0x7ff7e27e85d0)
21: gvk::dispatch_log (0x7ff7e27e8480)
20: gvk::context_vulkan::vk_debug_utils_callback (0x7ff7e2b24340)
19: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
18: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
17: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
16: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
15: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
14: vkNegotiateLoaderLayerInterfaceVersion (0x7ffdea8fe540)
13: CreateOrResizeBuffer (0x7ff7e2d43cd0)
12: ImGui_ImplVulkan_RenderDrawData (0x7ff7e2d47db0)
11: gvk::imgui_manager::render (0x7ff7e2c7d8d0)
10: gvk::sequential_invoker::execute_renders (0x7ff7e27a9190)
9: gvk::composition::render_thread (0x7ff7e27973b0)
8: fmt::v6::internal::basic_writer<fmt::v6::buffer_range<char> >::int_writer<unsigned __int64,fmt::v6::basic_format_specs<char> >::on_bin (0x7ff7e27b7de0)
7: gvk::start<gvk::physical_device_selection_hint,gvk::application_name,gvk::window * &,model_loader_app &,gvk::imgui_manager &> (0x7ff7e2991e40)
6: main (0x7ff7e27c8720)
```

There are also GitHub issues reporting this problem, but it appears like it can be fixed with the `MinImageCount` and `ImageCount` variables: https://github.com/ocornut/imgui/issues/3690

I'll also see if I can update ImGui to the latest version.

Reply:
Oh, didn't expect this to be different on AMD. I will check that out.

Reply:
ImGui has furthermore a hard-coded minimum (2) for MinImageCount. AMD GPU I have tested on has 1. The minimum has been set to two in all cases.